test/__snapshots__/type-generator.test.ts.snap

```diff
@@ -231,6 +231,86 @@ export enum FooFoo {
 "
 `;
 
+exports[`enums has allowlisted characters 1`] = `
+"/**
+ * @schema fqn.of.TestType
+ */
+export interface TestType {
+  /**
+   * @schema fqn.of.TestType#Chars
+   */
+  readonly chars?: FqnOfTestTypeChars;
+
+}
+
+/**
+ * Converts an object of type 'TestType' to JSON representation.
+ */
+/* eslint-disable max-len, quote-props */
+export function toJson_TestType(obj: TestType | undefined): Record<string, any> | undefined {
+  if (obj === undefined) { return undefined; }
+  const result = {
+    'Chars': obj.chars,
+  };
+  // filter undefined values
+  return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : ({ ...r, [i[0]]: i[1] }), {});
+}
+/* eslint-enable max-len, quote-props */
+
+/**
+ * @schema FqnOfTestTypeChars
+ */
+export enum FqnOfTestTypeChars {
+  /** != */
+  VALUE_NOT_EQUALS_TO = \\"!=\\",
+  /** ! */
+  VALUE_EXCLAMATION = \\"!\\",
+  /** <= */
+  VALUE_LESSER_THAN_EQUAL_TO = \\"<=\\",
+  /** !~ */
+  VALUE_NEGATION_TILDE = \\"!~\\",
+  /** ' */
+  VALUE_APOSTROPHE = \\"'\\",
+  /** \\\\ */
+  VALUE_BACKWARD_SLASH = \\"\\\\\\\\\\",
+  /** Foo.Bar!= */
+  FOO_PERIOD_BAR_NOT_EQUALS_TO = \\"Foo.Bar!=\\",
+  /** Foo.!=Bar */
+  FOO_PERIOD_NOT_EQUALS_TO_BAR = \\"Foo.!=Bar\\",
+  /** bool:true */
+  BOOL_TRUE = \\"bool:true\\",
+  /** Foo-Bar */
+  FOO_BAR = \\"Foo-Bar\\",
+  /** Bar_Foo */
+  BAR_FOO = \\"Bar_Foo\\",
+  /** 0.Foo */
+  VALUE_0_PERIOD_FOO = \\"0.Foo\\",
+  /** .Foo */
+  VALUE_PERIOD_FOO = \\".Foo\\",
+  /** Foo.0 */
+  FOO_PERIOD_0 = \\"Foo.0\\",
+  /** Foo. */
+  FOO_PERIOD = \\"Foo.\\",
+  /** 0.9 */
+  VALUE_0_9 = \\"0.9\\",
+  /** .9 */
+  VALUE_PERIOD_9 = \\".9\\",
+  /** 9 */
+  VALUE_9 = \\"9\\",
+  /** 9. */
+  VALUE_9_PERIOD = \\"9.\\",
```

Comment:
`Callout`:

This is a breaking change but I believe this is needed since it was not working and would give the same value.
Similar to these are,
```
Foo.
.Foo
```

For instance, if I run this with mainline, I see:
```
      | index.ts:39:3 - error TS2300: Duplicate identifier 'FOO'.
      | 39   FOO = ".Foo",
      |      ~~~
      | index.ts:43:3 - error TS2300: Duplicate identifier 'FOO'.
      | 43   FOO = "Foo.",
      |      ~~~
      | index.ts:47:3 - error TS2300: Duplicate identifier 'VALUE_9'.
      | 47   VALUE_9 = ".9",
      |      ~~~~~~~
      | index.ts:49:3 - error TS2300: Duplicate identifier 'VALUE_9'.
      | 49   VALUE_9 = "9",
      |      ~~~~~~~
      | index.ts:51:3 - error TS2300: Duplicate identifier 'VALUE_9'.
      | 51   VALUE_9 = "9.",
```